src/datasets/iterable_dataset.py

```diff
@@ -2074,7 +2075,10 @@ def from_generator(
                 Keyword arguments to be passed to the `generator` callable.
                 You can define a sharded iterable dataset by passing the list of shards in `gen_kwargs`.
                 This can be used to improve shuffling and when iterating over the dataset with multiple workers.
+            split(`NamedSplit`, default=Split.TRAIN):
```

Comment:
Nit (to be aligned with the other docstring; also you forgot to add a blank space after `split` and before `(`):
```suggestion
            split ([`NamedSplit`], defaults to `Split.TRAIN`)
```